### PR TITLE
Adjust appearance of priority notifications with buttons

### PR DESCRIPTION
**Pull request by thevoiceofzeke**
**In this PR:**
- Included Doug's work from the [MUMUP-2678 branch](https://github.com/UW-Madison-DoIT/uw-frame/tree/MUMUP-2678)
    - Added `actionButtons` attribute to sample notifications
    - `actionButtons` contains objects with the following attributes: `url`, `label`, `action`
- Added angular material attributes to ensure buttons appear in a row
- Added some CSS to reduce button size
- Changed background of priority notifications to generic light gray (previously we were lightening the theme's "color1," which was not always going to look good with white text)
- First button will be theme's primary color, last button will be the accent color
- Added black border around priority notifications to help set them apart from the rest of frame
- Added zero padding-left on `<ul>` lists to ensure consistency across browsers

### Screenshot
screen shot 2016-08-24 at 12 28 17 pm

### CODE REVIEW

**Comment by Doug-Reed** [comment]
:+1: 

That looks really good.

File: uw-frame-components/staticFeeds/notifications.json [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+        "title"  : "This is an admin notification smoke test.",
+        "actionURL" : "http://www.google.com",
+        "actionAlt" : "Google",
+        "dismissable" : false,
+        "priority" : true,
+		"buttonText" : [
+		  {"caption" : "Feedback", "url" : "/exclusive/feedback"},
+		  {"caption" : "Return To Old", "url" : "my.wisconsin.edu"}
+		]
+      }
```
Comment (timlevett) [inline uw-frame-components/staticFeeds/notifications.json]: Accidental? We do have the `sample_notifications.json` file in this same folder.

Comment (thevoiceofzeke) [inline uw-frame-components/staticFeeds/notifications.json]: I was going off of Doug's commits, where he moved his changes from sample into this file. I can move them back into sample if that's where they should be. I'm guessing it was moved here in preparation for [MUMUP-2765](https://jira.doit.wisc.edu/jira/browse/MUMMNG-2765). Is that right, @Doug-Reed?

File: uw-frame-components/css/buckyless/general.less [hunk 2]
```diff
@@ -28,6 +28,7 @@ p {
 ul {
   list-style-type: none;
   -webkit-padding-start: 0px;
+  paddding-left: 0;
```
Comment (timlevett) [inline uw-frame-components/css/buckyless/general.less]: What does this have to do with the priority notifications?

File: uw-frame-components/staticFeeds/notifications.json (hunk 1 above)
Comment (Doug-Reed) [inline uw-frame-components/staticFeeds/notifications.json]: This is accidental. This should come out.

**Comment by Doug-Reed** [comment]
@thevoiceofzeke : I'm going to delete the MUMUP-2678 branch, since this incorporates everything of value that was in there.

File: uw-frame-components/portal/notifications/partials/notification-bell.html [hunk 3]
```diff
@@ -31,7 +31,19 @@
 
   <!-- Priority notifications -->
   <div class="priority-notifications" ng-if='notificationsEnabled && (directiveMode === "priority")' ng-repeat="priority in priorityNotifications | limitTo: 1">
-    <a ng-href="{{priority.actionURL}}" alt="{{priority.actionAlt}}" ng-if='priorityNotifications.length == 1' class="notification-message">{{priority.title}}</a>
+    <div ng-if='priorityNotifications.length == 1' layout="row" layout-align="center center" layout-fill>
+      <a ng-href="{{priority.actionURL}}" alt="{{priority.actionAlt}}" class="notification-message">{{priority.title}}</a>
+      <div layout="row" layout-align="center center" ng-if='priority.actionButtons.length > 0' class="notification-buttons">
```
Comment (apetro) [inline uw-frame-components/portal/notifications/partials/notification-bell.html]: First check for `priority.actionButtons` to avoid null reference unhappiness. (Credit to @vertein for catching this.)

`priority.actionButtons && priority.actionButtons.length`

**Comment by vertein** [comment]
:+1:

**Comment by Doug-Reed** [comment]
:+1:

File: uw-frame-components/js/frame-config.js [hunk 4]
```diff
@@ -348,7 +348,7 @@ define(['angular'], function(angular) {
             }
           },
           {
-            "name" : "uw-oskhosh",
+            "name" : "uw-oshkosh",
```
Comment (timlevett) [inline uw-frame-components/js/frame-config.js]: If you change this name  also update the theme version, and change the theme file name in `uw-frame-components/css/themes`

File: uw-frame-components/staticFeeds/sample_notifications.json [hunk 5]
```diff
@@ -1,31 +1,37 @@
...
-      {
         "id"     : 2,
         "groups" : ["Users - Student SSN Update"],
         "title"  : "Our records indicate the Social Security Number in your Personal Information is invalid or missing. Please provide your Social Security Number or select to not provide your Social Security Number.",
         "actionURL" : "/portal/p/my-info-student-center-ssn.ctf1/max/action.uP?pP_action=loginAction",
         "actionAlt" : "UW-Madison Student Information",
         "dismissable" : true,
-        "priority" : false
+        "priority" : false,
+		"actionButtons" : []
```
Comment (timlevett) [inline uw-frame-components/staticFeeds/sample_notifications.json]: Are action buttons a required field for priority nots now? We should fail gracefully if it is null.

File: uw-frame-components/js/frame-config.js (hunk 4 above)
Comment (apetro) [inline uw-frame-components/js/frame-config.js]: This typo fix is a good catch; I think it deserves its own pull request (and its own JIRA issue to facilitate remembering to test this specifically as the software versions with this change migrate.)
